### PR TITLE
Feat: Numbas updated with polymorphic association

### PR DESCRIPTION
**Pull request by satikaj**
# Description

Latest commit introduces polymorphic association to task comments. Currently, task comments can belong to either Scorm test attempts or Overseer assessments.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Unit tests passed. Comments display as usual in the frontend.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation if appropriate
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have created or extended unit tests to address my new additions
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: db/migrate/20240601103707_add_polymorphic_association_to_comment.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+class AddPolymorphicAssociationToComment < ActiveRecord::Migration[7.1]
+  def change
+    remove_index :task_comments, :overseer_assessment_id
+
+    add_column :task_comments, :commentable_type, :string
+    rename_column :task_comments, :overseer_assessment_id, :commentable_id
+
+    TaskComment.find_each do |comment|
+      if comment.commentable_id.present?
```
Comment (macite) [inline db/migrate/20240601103707_add_polymorphic_association_to_comment.rb]: You should be able to do this with a where - TaskComment.where('NOT commendable_id IS NULL').find_each

File: db/migrate/20240601103707_add_polymorphic_association_to_comment.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+        comment.update(commentable_type: 'OverseerAssessment')
```
Comment (macite) [inline db/migrate/20240601103707_add_polymorphic_association_to_comment.rb]: I think you could then do update all... something like this:

```suggestion
     TaskComment.where('NOT commendable_id IS NULL').in_batches.update_all(commentable_type: 'OverseerAssessment')
```

**Review by macite — CHANGES_REQUESTED** [review]
Looking super close - the Task Comment table is huge, so I think we can optimise the update... see the comment. @satikaj do you have time to make these changes?

File: db/migrate/20240601103707_add_polymorphic_association_to_comment.rb (hunk 1 above)
Comment (satikaj) [reply db/migrate/20240601103707_add_polymorphic_association_to_comment.rb]: I've made the changes now, Andrew

**Pull request merged**